test/functional/test_framework/wallet.py

```diff
@@ -67,13 +67,26 @@ def send_self_transfer(self, *, fee_rate=Decimal("0.003"), from_node, utxo_to_sp
         tx = CTransaction()
         tx.vin = [CTxIn(COutPoint(int(utxo_to_spend['txid'], 16), utxo_to_spend['vout']))]
         tx.vout = [CTxOut(int(send_value * COIN), self._scriptPubKey)]
+        tx.nLockTime = locktime
         tx.wit.vtxinwit = [CTxInWitness()]
         tx.wit.vtxinwit[0].scriptWitness.stack = [CScript([OP_TRUE])]
         tx_hex = tx.serialize().hex()
 
         tx_info = from_node.testmempoolaccept([tx_hex])[0]
-        self._utxos.append({'txid': tx_info['txid'], 'vout': 0, 'value': send_value})
+        if tx_info['allowed']:
+            assert_equal(tx_info['vsize'], vsize)
+            assert_equal(tx_info['fees']['base'], fee)
+
+        new_utxo = {'txid': tx_info['txid'], 'vout': 0, 'value': send_value}
+        return tx_hex,  new_utxo
```

Comment:
I'm not very sure the intuition behind this but, can we just add everything in one variable and use key/value pairs to retrieve things? Seems cleaner to me and wouldn't have to include dead `_` every time I create a tx. For instance instead of doing `spend_101_raw, _ = wallet.create_self_transfer(from_node=self.nodes[0], utxo_to_spend=utxo_101)` we can simplt do       `spend_101_raw= wallet.create_self_transfer(from_node=self.nodes[0], utxo_to_spend=utxo_101)['tx_hex']`

Reply:
done!